lib/src/vector_math/matrix4.dart

```diff
@@ -684,17 +684,14 @@ class Matrix4 {
     } else if (x is Vector4) {
       translateByVector4(x);
     } else if (x is double) {
-      translateByDouble(x, y, z, 1.0);
+      translateByDouble(x, y, z);
     } else {
       throw UnimplementedError();
     }
   }
 
   /// Translate this matrix by x, y, z, w.
-  @pragma('wasm:prefer-inline')
-  @pragma('vm:prefer-inline')
-  @pragma('dart2js:prefer-inline')
-  void translateByDouble(double tx, double ty, double tz, double tw) {
+  void translateByDouble(double tx, double ty, double tz, [double tw = 1.0]) {
```

Comment:
Optional parameters compliate calling conventions, the function now needs to check every time whether `tw` was called or not. It's a good idea to avoid these in performance sensitive code, though I have no evidence that it matters in this particular case. We should benchmark and measure.

You can observe the difference by compiling to Wasm before and after and checking the code for this function. Make sure to call it with and without the optional parameter otherwise TFA will optimize it into a version that always takes `tw` or never takes it.

Alternatively you can inline it always, in which case you get optimized code with no "is tw passed" checks, as in each call site you'll either pass it or not, and generate the code based on that.

Reply:
Should I drop em? Please give me code 😄

Reply:
Didn't I give you the code already in first round of review?

This usage of optional parameter isn't going to improve things - most likely the compiler will just make the call site implicitly pass `1.0` where before your changes this was done explicitly.

So it suffers from the same problem I outlined before, that even though a parameter is `1.0` constant, we don't inline, so we do unnecessary extra work.